pkg/callback/callback.go

```diff
@@ -0,0 +1,180 @@
+	Callback(ctx context.Context, pl map[string]interface{}) error
+}
+
+// NewCallback creates a new instance of CallBack.
+func NewCallback(httpClient *http.Client, method string, urls []string, permissions []string) (Callback, error) {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+	if method != http.MethodPost && method != http.MethodGet {
+		return nil, fmt.Errorf("unsupported auth callback method: %s", method)
+	}
+
+	allowedPermission := make(map[string]struct{})
+	for _, permission := range permissions {
+		allowedPermission[permission] = struct{}{}
+	}
+
+	return &callback{
+		httpClient:        httpClient,
+		urls:              urls,
+		method:            method,
+		allowedPermission: allowedPermission,
+	}, nil
+}
+
+func NewCalloutClient(ctls bool, certPath, keyPath, caPath string, timeout time.Duration) (*http.Client, error) {
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: !ctls,
+	}
+	if certPath != "" || keyPath != "" {
+		clientTLSCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		if err != nil {
+			return nil, err
+		}
+		certPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+		caCert, err := os.ReadFile(caPath)
+		if err != nil {
+			return nil, err
+		}
+		if !certPool.AppendCertsFromPEM(caCert) {
+			return nil, errors.Wrap(errors.New("failed to append CA certificate"), svcerr.ErrCreateEntity)
+		}
+		tlsConfig.RootCAs = certPool
+		tlsConfig.Certificates = []tls.Certificate{clientTLSCert}
+	}
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+		Timeout: timeout,
+	}
+
+	return httpClient, nil
+}
+
+func (c *callback) makeRequest(ctx context.Context, urlStr string, params map[string]string) error {
+	var req *http.Request
+	var err error
+
+	switch c.method {
+	case http.MethodGet:
+		query := url.Values{}
+		for key, value := range params {
+			query.Set(key, value)
+		}
+		req, err = http.NewRequestWithContext(ctx, c.method, urlStr+"?"+query.Encode(), nil)
+	case http.MethodPost:
+		data, jsonErr := json.Marshal(params)
+		if jsonErr != nil {
+			return jsonErr
+		}
+		req, err = http.NewRequestWithContext(ctx, c.method, urlStr, bytes.NewReader(data))
+		req.Header.Set("Content-Type", "application/json")
+	}
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.NewSDKErrorWithStatus(svcerr.ErrAuthorization, resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (c *callback) Callback(ctx context.Context, pl map[string]interface{}) error {
+	if len(c.urls) == 0 {
+		return nil
+	}
+
+	// Check if the permission is in the allowed list
+	// Otherwise, only call webhook if the permission is in the map
+	if len(c.allowedPermission) > 0 {
+		val, ok := pl["permission"].(string)
+		if ok {
+			_, exists := c.allowedPermission[val]
+			if !exists {
+				return nil
+			}
+		}
+	}
+
```

Comment:
This logic no needed, This function just need to send what ever is given

```suggestion
-	if len(c.allowedPermission) > 0 {
-		val, ok := pl["permission"].(string)
-		if ok {
-			_, exists := c.allowedPermission[val]
-			if !exists {
-				return nil
-			}
-		}
-	}
```